File: src/main/java/nu/peg/slack/pt/api/slack/model/ActionAttachment.java
package nu.peg.slack.pt.api.slack.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.google.gson.annotations.SerializedName;
import lombok.Data;

import java.util.List;

@Data
public class ActionAttachment implements Attachment {

    private String title;

    private String text;

    @SerializedName("callback_id")
    @JsonProperty("callback_id")
    private String callbackId;

    private String color;

    private List<Action> actions;

}

File: src/main/kotlin/nu/peg/slack/pt/api/slack/model/ActionAttachment.kt
package nu.peg.slack.pt.api.slack.model

import com.google.gson.annotations.SerializedName

import com.fasterxml.jackson.annotation.JsonProperty

import lombok.Data

@Data
class ActionAttachment : Attachment {

    private val title: String? = null

    private val text: String? = null

    @SerializedName("callback_id")
    @JsonProperty("callback_id")
    private val callbackId: String? = null

    private val color: String? = null

    private val actions: List<Action>? = null

}
